### ISSUE TITLE
[Day 3] Dense 신경망으로 식물 병리 분류 완료 - {zeenee}

### ISSUE THREAD

**Issue by jinnnih** [jinnnih/Algorithm_Study#9]
## 실습 완료 내용

- [x] Plant Pathology 2020 데이터 로드 & 탐색

- [x] 이미지 전처리 (224×224 축소, 정규화)

- [x] Dense 신경망 구축 (4-class 분류)

- [x] 클래스 불균형 보정 (class_weight)

- [x] 모델 훈련 (조기 종료 적용)

- [x] 성능 평가 및 혼동행렬 분석

## 데이터셋 정보

- 원본 크기: 1365×1024 이미지

- 전처리: 224×224 축소 (메모리 효율)

- 클래스 분포:

  - Healthy: 516장 (28.3%)

  - Multiple Diseases: 91장 (5.0%) <- 주의: 매우 적음

  - Rust: 622장 (34.2%)

  - Scab: 592장 (32.5%)

<img width="632" height="717" alt="Image" src="https://github.com/user-attachments/assets/e98d42d9-9bf7-4d73-af6a-98af6e640fbb" />

<img width="546" height="422" alt="Image" src="https://github.com/user-attachments/assets/2441d0e1-f694-47b4-a3d8-f478d133af2e" />

<img width="296" height="289" alt="Image" src="https://github.com/user-attachments/assets/7750da3f-2d61-4f57-b19e-f20593a437cc" />

<img width="657" height="550" alt="Image" src="https://github.com/user-attachments/assets/f7c1f73c-a6f5-4a3c-9925-050f072dd356" />

<img width="622" height="514" alt="Image" src="https://github.com/user-attachments/assets/b7583e42-005d-44d7-acd3-1ffeae8702f3" />

<img width="518" height="602" alt="Image" src="https://github.com/user-attachments/assets/fca4e5f5-925b-4557-85d0-80fb67ad831e" />

## 모델 구축

- 입력층: 150,528 (224×224×3 이미지 flatten)

- 은닉층 1: Dense(512) + BatchNorm + Dropout(0.4) + ReLU

- 은닉층 2: Dense(256) + BatchNorm + Dropout(0.3) + ReLU

- 은닉층 3: Dense(128) + BatchNorm + Dropout(0.2) + ReLU

- 은닉층 4: Dense(64) + ReLU

- 출력층: Dense(4) + Softmax (다중분류)

## 성능 결과

- 훈련 정확도:  86.33% (최종 에포크 기준)

- 검증 정확도:  44.38% (목표 70% 미달성)

- Log Loss: 1.2533 (낮을수록 좋음)

- 훈련 에포크: 23개 (EarlyStopping 적용)

## 혼동행렬 분석 (4×4)

실제 \ 예측 | Healthy | Multiple | Rust | Scab
-- | -- | -- | -- | --
Healthy | 51 | 6 | 19 | 27
Multiple | 1 | 4 | 7 | 6
Rust | 33 | 7 | 48 | 37
Scab | 18 | 10 | 33 | 58
(※ 분류 리포트의 recall과 혼동 분석 데이터를 종합하여 작성되었습니다.) |   |   |   |  

## 클래스별 정확도

- Healthy:  50% (재현율)

- Multiple Diseases: 22% (가장 낮은 성능, 데이터 부족 영향)

- Rust: 38% 

- Scab: 49% 

## 관찰 기록

1. 이미지 축소 (1365×1024 → 224×224)가 성능에 미친 영향? 
고해상도를 224×224로 대폭 축소하면서 미세한 질병 특징이 소실되었고, 
Dense 레이어 특성상 위치 정보를 무시하여 성능이 낮게 나옴.

2. 클래스 가중치 적용 전후 비교: 
적용시 Multiple Diseases에 약 4.98의 높은 가중치를 부여하여 학습을 유도했으나, 
절대적인 샘플 수 부족으로 인해 여전히 낮은 정확도(22%)를 보임.

3. 가장 많이 혼동되는 클래스 쌍:
Rust ↔ Scab. 
두 클래스 모두 잎에 얼룩이 있는 형태로, 
공간 정보 없이 픽셀 값만 학습하는 Dense 신경망으로는 구분이 매우 어려움

## 개선 시도 (선택)

- 문제점 : 훈련 정확도(86%)에 비해 검증 정확도(44%)가 현저히 낮아 과적합 발생

- 가장 효과 있었던 변경: 이미지 데이터의 공간적 특징을 보존할 수 있는 CNN(합성곱 신경망) 도입이 절실함

## 다음 단계 (내일)

CNN으로 같은 문제를 풀어보고, 이미지의 '공간 정보'를 유지하는 것이 성능에 어떤 영향을 주는지 비교 분석 예정

## 커밋 및 링크

- Kaggle 노트북: (URL)https://www.kaggle.com/code/owojini/plant-pathology-2020-fgvc7-practice

- GitHub: (링크) https://github.com/jinnnih/Algorithm_Study/issues/9
